assets/app/scripts/controllers/builds.js

```diff
@@ -58,6 +94,61 @@ angular.module('openshiftConsole')
       }      
     };
 
+    // Function which will 'instantiate' new build from given buildConfigName
+    $scope.startBuild = function(buildConfigName) {
+      var req = {metadata:{name:buildConfigName}};
+      DataService.create("buildConfigs/instantiate", buildConfigName, req, $scope).then(
+        function(build) { //success
+            $scope.alerts = [
+            {
+              type: "success",
+              message: "Build " + build.metadata.name + " has started.",
+            }
+          ];
+        },
+        function(result) { //failure
+          $scope.alerts = [
+            {
+              type: "error",
+              message: "An error occurred while starting the build.",
+              details: "Status: " + result.status + ". " + result.data,
+            }
+          ];
+        }
+      );
+    };
+
+    // Function which will 'clone' build from given buildName
+    $scope.cloneBuild = function(buildName) {
+      var req = {metadata:{name:buildName}};
+      DataService.create("builds/clone", buildName, req, $scope).then(
+        function(build) { //success
+            $scope.alerts = [
+            {
+              type: "success",
+              message: "Build " + buildName + " is being rebuilt as " + build.metadata.name + ".",
+            }
+          ];
+        },
+        function(result) { //failure
+          $scope.alerts = [
+            {
+              type: "error",
+              message: "An error occurred while rerunning the build.",
+              details: "Status: " + result.status + ". " + result.data,
+            }
+          ];
+        }
+      );
+    };
+
+    // Function to disable Start/Rebuild build button, which checks,
+    // if any other build from given buildConfig is running.
+    // Returns true/false
+    $scope.isBuildRunning = function(buildConfigName) {
+      return (Object.getOwnPropertyNames($scope.buildConfigBuildsInProgress[buildConfigName]).length !== 0);
```

Comment:
we have a filter that does this already, you can include the hashSizeFilter as a dependency in the controller and then call it like `return hashSizeFilter($scope.buildConfigBuildsInProgress[buildConfigName]) > 0`

alternatively in the actual template you can do `{{($scope.buildConfigBuildsInProgress[buildConfigName] | hashSize) > 0}}`  without even needing to include the filter as a dependency.